indexer/helper/evm.go

```diff
@@ -0,0 +1,465 @@
+				},
+			},
+			Schema: "JsonRPC",
+		},
+		"/blockReceipts": {
+			QueryParameter: []types.ParameterIndex{
+				{
+					IndexId:     utils.IndexEVMReceipt,
+					Parameter:   []string{"hash"},
+					Description: []string{"hash of a block"},
+				},
+			},
+			Schema: "JsonRPC",
+		},
+		//"/logs": {
+		//	QueryParameter: []types.ParameterIndex{
+		//		{
+		//			IndexId:     utils.IndexEVMLog,
+		//			Parameter:   []string{"address", "blockHash"},
+		//			Description: []string{"contract address or a list of addresses from which logs should originate", "EVM block hash"},
+		//		},
+		//	},
+		//	Schema: "JsonRPC",
+		//},
+	}
+}
+
+type EVMDataItem struct {
+	Block struct {
+		Hash                 string            `json:"hash"`
+		ParentHash           string            `json:"parentHash"`
+		Number               int64             `json:"number"`
+		Timestamp            int64             `json:"timestamp"`
+		Nonce                string            `json:"nonce"`
+		Difficulty           int64             `json:"difficulty"`
+		GasLimit             json.RawMessage   `json:"gasLimit"`
+		GasUsed              json.RawMessage   `json:"gasUsed"`
+		Miner                string            `json:"miner"`
+		ExtraData            string            `json:"extraData"`
+		Transactions         []json.RawMessage `json:"transactions"`
+		BaseFeePerGas        json.RawMessage   `json:"baseFeePerGas"`
+		UnderscoreDifficulty json.RawMessage   `json:"_difficulty"`
+	} `json:"block"`
+	Receipts []Receipt `json:"receipts"`
+}
+
+type EVMDataItemRaw struct {
+	Block    json.RawMessage   `json:"block"`
+	Receipts []json.RawMessage `json:"receipts"`
+}
+
+type Receipt struct {
+	Status            string            `json:"status"`
+	CumulativeGasUsed string            `json:"cumulativeGasUsed"`
+	Logs              []json.RawMessage `json:"logs"`
+	LogsBloom         string            `json:"logsBloom"`
+	Type              string            `json:"type"`
+	TransactionHash   string            `json:"transactionHash"`
+	TransactionIndex  string            `json:"transactionIndex"`
+	BlockHash         string            `json:"blockHash"`
+	BlockNumber       string            `json:"blockNumber"`
+	GasUsed           string            `json:"gasUsed"`
+	EffectiveGasPrice string            `json:"effectiveGasPrice"`
+	From              string            `json:"from"`
+	To                *string           `json:"to"`
+	ContractAddress   *string           `json:"contractAddress"`
+}
+
+type Transaction struct {
+	Hash string `json:"hash"`
+}
+
+type Log struct {
+	Address         string `json:"address"`
+	BlockHash       string `json:"blockHash"`
+	LogIndex        string `json:"logIndex"`
+	TransactionHash string `json:"transactionHash"`
+}
+
+func (c *EVMIndexer) calculateMerkleRoot(item interface{}) [32]byte {
+	var leafs [][32]byte
+
+	switch v := item.(type) {
+	case *[]json.RawMessage:
+		leafs = make([][32]byte, 0, len(*v))
```

Comment:
runtime reflection is pretty slow here. We should use a generic function for this. Something like:

```go
func getMerkleRoot[T any](array *[]T) *[][32]byte {
	leafs := make([][32]byte, 0, len(*array))
	for _, item := range *array {
		leafs = append(leafs, utils.CalculateSHA256Hash(item))
	}
	return merkle.GetMerkleRoot(leafs)
}
```